### DISCUSSION TITLE
UI Overlap Mitigation?

### DISCUSSION THREAD

**Discussion by LinuxinaBit** [Akylas/oss-weather#238] [General]
Is there a chance of using some kind of CSS flexbox-like technology (plus some related max element size limts) to mitigate the various UI overlap issues?

It could also help fix the various places that spacing inconsistencies show up if implemented properly...

Of course actual flexboxes probably aren't an option (this isn't a website of course), but there's gotta be something similar available.

![css-flexbox-poster](https://github.com/user-attachments/assets/7d6a1008-5dd5-4686-9eb5-b24ca4077d05)

**Comment by farfromrefug** [comment]
Actually it is possible. But what ui changes are you looking for ?

**Comment by LinuxinaBit** [comment]
Mainly there are problems with the weekly view icons at normal font scales; they need better centering in between the temperatures and the date (instead of being directly in the middle of the screen), and the list should probably become scrollable instead of overlapping with other elements when icon count increases (if possible).

I think it would also be sensible to put some limits in place to prevent overlap when font sizes are increased as well, but that's a much more involved effort requiring relative size limits on everything.

Of course these are relatively minor nitpicks of a very useful and amazing app, great job as always ❤️

**Reply by farfromrefug** [reply]
@LinuxinaBit yes it is a tricky subject. I dont really like the scrolling way because it is not really nice in a vertical llistview.
now yes that list can get too big. This is why i created the `small icon weather data` where some icons can be shown as small icon on the bottom left (i use it for moon, aqi, uv where color/icon is enough). There is also the "line" design allowing to show around 8 datas at the same time without overlapping with data/temp.

I will add clipping though to ensure it does not look bad (too bad...)
